data_diff/__init__.py

```diff
@@ -138,5 +144,15 @@ def diff_tables(
         )
     else:
         raise ValueError(f"Unknown algorithm: {algorithm}")
+    
+    # no key_columns provided, use table segment key_columns
+    # filter to unique values
+    if key_columns is None:
+        key_columns = list(set(list(segments[0].key_columns + segments[1].key_columns)))
```

Comment:
This is a bug.

We don't actually care about the key_columns, print_stats only wants to know *how many* key columns we have (i.e. 1 or more), and that has to be the same for both segments.

Reply:
https://github.com/datafold/data-diff/pull/300/commits/2f709c52b8b2866730e92124b1a4fe4d413a16a8